uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/main/java/org/guvnor/m2repo/backend/server/M2ServletContextListener.java

```diff
@@ -0,0 +1,198 @@
+    private final String ARTIFACT_ID = "artifactId";
+    private final String VERSION = "version";
+    private final String POM_PROPERTIES = "pom.properties";
+    private String JARS_FOLDER = File.separator + WEB_INF_FOLDER + File.separator + LIB_FOLDER + File.separator;
+    private String MAVEN_META_INF = "META-INF" + File.separator + "maven";
+
+    private Logger logger = LoggerFactory.getLogger(M2ServletContextListener.class);
+
+    public M2ServletContextListener() {
+    }
+
+    @Override
+    public void contextInitialized(ServletContextEvent servletContextEvent) {
+        logger.info("M2ServletContextListener contextInitialized started");
+        ServletContext ctx = servletContextEvent.getServletContext();
+        String jarsPath = ctx.getRealPath(JARS_FOLDER);
+        long startTime = System.nanoTime();
+        int jarsDeployed = deployJarsFromWar(jarsPath);
+        long endTime = System.nanoTime();
+        long totalTime = TimeUnit.NANOSECONDS.toSeconds(endTime - startTime);
+        logger.info("M2ServletContextListener contextInitialized deployed {} jars in {} sec ",
+                    jarsDeployed,
+                    totalTime);
+    }
+
+    @Override
+    public void contextDestroyed(ServletContextEvent servletContextEvent) {
+    }
+
+    private int deployJarsFromWar(String path) {
+        int i = 0;
+        try (DirectoryStream<Path> ds = Files.newDirectoryStream(Paths.get(path))) {
+            RepositorySystemSession session = newSession(newRepositorySystem());
+            for (Path p : ds) {
+                if (p.toString().endsWith(JAR_EXT)) {
+                    deployJar(p.toAbsolutePath().toString(),
+                              session);
+                    i++;
+                }
+            }
+        } catch (Exception e) {
+            logger.error(e.getMessage(),
+                         e);
+        }
+        return i;
+    }
+
+    public GAV deployJar(String file,
+                         RepositorySystemSession session) {
+        GAV gav = new GAV();
+        Properties props = readZipFile(file);
+        if (!props.isEmpty()) {
+            gav = new GAV(props.getProperty(GROUP_ID),
+                          props.getProperty(ARTIFACT_ID),
+                          props.getProperty(VERSION));
+            deploy(gav,
+                   file,
+                   session);
+        }
+        return gav;
+    }
+
+    public Properties readZipFile(String zipFilePath) {
+        try {
+            ZipFile zipFile = new ZipFile(zipFilePath);
+            Enumeration<? extends ZipEntry> e = zipFile.entries();
+            while (e.hasMoreElements()) {
+                ZipEntry entry = e.nextElement();
+                String end = POM_PROPERTIES;
+                if (!entry.isDirectory() && entry.getName().startsWith(MAVEN_META_INF) && entry.getName().endsWith(end)) {
+                    BufferedInputStream bis = new BufferedInputStream(zipFile.getInputStream(entry));
```

Comment:
Please use try with resources to guarantee stream closing in case of exceptions. Code with all objections fixed:

```java
    private Properties readZipFile(String zipFilePath) {
        try {
            ZipFile zipFile = new ZipFile(zipFilePath);
            Enumeration<? extends ZipEntry> e = zipFile.entries();
            while (e.hasMoreElements()) {
                ZipEntry entry = e.nextElement();
                if (!entry.isDirectory()
                        && entry.getName().startsWith(MAVEN_META_INF)
                        && entry.getName().endsWith(POM_PROPERTIES)) {
                    try (BufferedInputStream bis = new BufferedInputStream(zipFile.getInputStream(entry))) {
                        Properties props = new Properties();
                        props.load(bis);
                        return props;
                    }
                }
            }
        } catch (IOException e) {
            logger.error("IOError :{}",
                         e.getMessage(),
                         e);
        }
        return new Properties();
    }
```

Reply:
FIxed thanks !